docs/assemblies/con_provisioning-bare-metal-data-plane-nodes.adoc

```diff
@@ -163,6 +163,160 @@ When bonding is configured, the `ctlplaneInterface` should be set to the bond in
+
+=== Building a Container Image from an Existing qcow2
+
+If you have an existing qcow2 image, you can package it using the `copy_out.sh` script from
+the https://github.com/openstack-k8s-operators/edpm-image-builder[edpm-image-builder]
+repository.
+
+==== Step 1: Generate Checksum File
+
+*Required:* Create a checksum file for your qcow2 image. The provision server requires a
+checksum file to function properly - the checksum discovery agent will fail if no checksum
+file is found. The provision server supports MD5, SHA256, or SHA512 checksums. The checksum
+file must contain the hash type in its filename (e.g., `md5`, `sha256`, or `sha512`):
+
+[,sh]
+----
+# For SHA256 (recommended)
+sha256sum my-custom-image.qcow2 > my-custom-image.qcow2.sha256sum
+
+# Or for MD5
+md5sum my-custom-image.qcow2 > my-custom-image.qcow2.md5sum
+
+# Or for SHA512
+sha512sum my-custom-image.qcow2 > my-custom-image.qcow2.sha512sum
+----
+
+==== Step 2: Clone the Repository
+
+Clone the edpm-image-builder repository to get the `copy_out.sh` script:
+
+[,sh]
+----
+git clone https://github.com/openstack-k8s-operators/edpm-image-builder.git
+cd edpm-image-builder
+----
+
+==== Step 3: Create Containerfile
+
+Create a `Containerfile` (or `Dockerfile`) in the same directory. Copy both your qcow2 image
+and its checksum file (checksum is required):
+
+[,dockerfile]
+----
+FROM registry.access.redhat.com/ubi9/ubi-minimal:9.6
+
+# Copy your qcow2 image and checksum file into the container
+# The copy_out.sh script expects files in the root directory (/) by default
+COPY my-custom-image.qcow2 /
+COPY my-custom-image.qcow2.sha256sum /
+
+# Copy the copy_out.sh script from the repository
+COPY copy_out.sh /copy_out.sh
+RUN chmod +x /copy_out.sh
+
+ENTRYPOINT ["/copy_out.sh"]
+----
+
+*Note:*
+
+* Replace `my-custom-image.qcow2` with your actual qcow2 filename
+* Replace `my-custom-image.qcow2.sha256sum` with your checksum filename (must contain `md5`,
+`sha256`, or `sha512` in the filename to be detected by the provision server)
+* The files are copied to `/` (root directory) because `copy_out.sh` expects to find them
+there by default (the default `SRC_DIR` is `/`). You can set `ENV SRC_DIR=<path>` in your
+Containerfile if you want to use a different source directory.
+* The `copy_out.sh` script handles both compressed (`.qcow2.gz`) and uncompressed (`.qcow2`)
+images
+
+==== Step 4: Build and Push
+
+Build and push the container image:
+
+[,sh]
+----
+buildah bud -f Containerfile -t <your-registry>/my-custom-os-image:latest
+buildah push <your-registry>/my-custom-os-image:latest
+----
+
+Or using podman/docker:
+
+[,sh]
+----
+podman build -f Containerfile -t <your-registry>/my-custom-os-image:latest
+podman push <your-registry>/my-custom-os-image:latest
+----
+
+=== Using a Custom Image with OpenStackProvisionServer
+
+After building and pushing your container image, use it with the `OpenStackProvisionServer`
+resource:
```

Comment:
Why is it needed to use a custom ProvisionServer?
The qcow2 container can be specified under osContainerImageUrl directly on baremetalSetTemplate, and that will automatically create a ProvisionServer serving that image.

Reply:
Yeah, there are multiple options available here. The best option is to patch openstackversion so that it can be used across nodesets without patching them separately. The custom provisionserver (can be used with multiple nodesets) is useful  as one can exhaust the provisionserver ports, if we create one for every nodeset which is the default. I'll update the docs to list the three options.